local-tests/tests/testDelegatingCapacityCreditsNFTToAnotherPkpToExecuteJs.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  AuthMethodScope,
+  AuthMethodType,
+  LIT_ENDPOINT_VERSION,
+} from '@lit-protocol/constants';
+import { LitAuthClient } from '@lit-protocol/lit-auth-client';
+import { LitActionResource, LitPKPResource } from '@lit-protocol/auth-helpers';
+import { LitAbility } from '@lit-protocol/types';
+import { TinnyEnvironment } from 'local-tests/setup/tinny-environment';
+import { LIT_TESTNET } from 'local-tests/setup/tinny-config';
+
+/**
+ * ## Scenario:
+ * Delegating capacity credits NFT to Bob (delegatee) for him to execute JS code to sign with his PKP
+ * - Given: The capacity credits NFT is minted by the dApp owner
+ * - When: The dApp owner creates a capacity delegation authSig
+ * - And: The dApp owner delegates the capacity credits NFT to Bob
+ * - Then: The delegated (Bob's) wallet can execute JS code to sign with his PKP using the capacity from the capacity credits NFT
+ *
+ *
+ * ## Test Commands:
+ * - ❌ Not supported in Cayenne
+ * - ✅ NETWORK=manzano yarn test:local --filter=testDelegatingCapacityCreditsNFTToAnotherPkpToExecuteJs
+ * - ✅ NETWORK=localchain yarn test:local --filter=testDelegatingCapacityCreditsNFTToAnotherPkpToExecuteJs
+ */
+export const testDelegatingCapacityCreditsNFTToAnotherPkpToExecuteJs = async (
+  devEnv: TinnyEnvironment
+) => {
+  devEnv.setUnavailable(LIT_TESTNET.CAYENNE);
+
+  const alice = await devEnv.createRandomPerson();
+  const bob = await devEnv.createRandomPerson();
+
+  // Checking the scopes of the PKP owned by Bob
+  const bobsAuthMethodAuthId = await LitAuthClient.getAuthIdByAuthMethod(
+    bob.authMethod
+  );
+
+  const scopes =
+    await bob.contractsClient.pkpPermissionsContract.read.getPermittedAuthMethodScopes(
+      bob.authMethodOwnedPkp.tokenId,
+      AuthMethodType.EthWallet,
+      bobsAuthMethodAuthId,
+      3
+    );
+
+  if (!scopes[AuthMethodScope.SignAnything]) {
+    throw new Error('Bob does not have the "SignAnything" scope on his PKP');
+  }
+
+  // As a dApp owner, create a capacity delegation authSig for Bob's PKP wallet
+  const capacityDelegationAuthSig = await alice.createCapacityDelegationAuthSig(
+    [bob.pkp.ethAddress]
+  );
+
+  // As a dApp owner, delegate the capacity credits NFT to Bob
+  const bobPkpSessionSigs = await devEnv.litNodeClient.getPkpSessionSigs({
+    pkpPublicKey: bob.authMethodOwnedPkp.publicKey,
+    authMethods: [bob.authMethod],
+    resourceAbilityRequests: [
+      {
+        resource: new LitPKPResource('*'),
+        ability: LitAbility.PKPSigning,
+      },
+      {
+        resource: new LitActionResource('*'),
+        ability: LitAbility.LitActionExecution,
+      },
+    ],
+    capabilityAuthSigs: [capacityDelegationAuthSig],
```

Comment:
You're testing it against only the local nodes since Manzano won't have these changes? But local nodes don't enforce capacity credits so we aren't really testing this?